packages/editor/CodeMirror/editorCommands/duplicateLine.ts

```diff
@@ -0,0 +1,32 @@
+import { EditorSelection } from '@codemirror/state';
+import { Command, EditorView } from '@codemirror/view';
+
+const duplicateLine: Command = (editor: EditorView) => {
+	const state = editor.state;
+	const doc = state.doc;
+
+	const transaction = state.changeByRange(range => {
+		const currentLine = doc.lineAt(range.anchor);
+
+		let text, insertPos;
+		if (range.empty) {
+			text = `\n${currentLine.text}`;
+			insertPos = currentLine.to;
+		} else {
+			text = doc.slice(range.from, range.to);
+			insertPos = range.to;
+		}
+		return {
+			range: EditorSelection.range(range.to, range.to + text.length),
```

Comment:
In the CodeMirror 5 editor, it seems to leave the selection as just a single cursor when there is no selection. It might make sense to duplicate that behavior here.

Reply:
done.